autogpt_platform/backend/backend/blocks/airtable/_webhook.py

```diff
@@ -0,0 +1,155 @@
+"""
+Webhook management for Airtable blocks.
+"""
+
+import hashlib
+import hmac
+import logging
+from enum import Enum
+from typing import Tuple
+
+from backend.sdk import (
+    BaseWebhooksManager,
+    Credentials,
+    ProviderName,
+    Webhook,
+    update_webhook,
+)
+
+from ._api import (
+    WebhookFilters,
+    WebhookSpecification,
+    create_webhook,
+    delete_webhook,
+    list_webhook_payloads,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class AirtableWebhookEvent(str, Enum):
+    TABLE_DATA = "tableData"
+    TABLE_FIELDS = "tableFields"
+    TABLE_METADATA = "tableMetadata"
+
+
+class AirtableWebhookManager(BaseWebhooksManager):
+    """Webhook manager for Airtable API."""
+
+    PROVIDER_NAME = ProviderName("airtable")
+
+    @classmethod
+    async def validate_payload(
+        cls, webhook: Webhook, request, credentials: Credentials | None
+    ) -> Tuple[dict, str]:
+        """Validate incoming webhook payload and signature."""
+
+        if not credentials:
+            raise ValueError("Missing credentials in webhook metadata")
+
+        payload = await request.json()
+
+        # Verify webhook signature using HMAC-SHA256
+        if webhook.secret:
+            mac_secret = webhook.config.get("mac_secret")
+            if mac_secret:
+                # Get the raw body for signature verification
+                body = await request.body()
+
+                # Calculate expected signature
+                mac_secret_decoded = mac_secret.encode()
+                hmac_obj = hmac.new(mac_secret_decoded, body, hashlib.sha256)
+                expected_mac = f"hmac-sha256={hmac_obj.hexdigest()}"
+
+                # Get signature from headers
+                signature = request.headers.get("X-Airtable-Content-MAC")
+
+                if signature and not hmac.compare_digest(signature, expected_mac):
+                    raise ValueError("Invalid webhook signature")
+
+        # Validate payload structure
+        required_fields = ["base", "webhook", "timestamp"]
+        if not all(field in payload for field in required_fields):
+            raise ValueError("Invalid webhook payload structure")
+
+        if "id" not in payload["base"] or "id" not in payload["webhook"]:
+            raise ValueError("Missing required IDs in webhook payload")
+        base_id = payload["base"]["id"]
+        webhook_id = payload["webhook"]["id"]
+
+        # get payload request parameters
+        cursor = webhook.config.get("cursor", 1)
+
+        response = await list_webhook_payloads(credentials, base_id, webhook_id, cursor)
+
+        # update webhook config
+        await update_webhook(
+            webhook.id,
+            config={"base_id": base_id, "cursor": response.cursor},
+        )
+
+        event_type = "notification"
+        return response.model_dump(), event_type
```

Comment:
In this case I'd suggest changing the signature of `validate_payload` to allow returning a list of payloads. Otherwise multiple (possibly unrelated) payloads can end up in the same graph execution, which I don't think is desirable.

This can be fixed in a follow-up PR, I just wanted to post it here for your consideration.